### ISSUE TITLE
cannot forward to af 2, proto 1 no corresponding listening socket

### ISSUE THREAD

**Issue by denser** [xlab1/sipfe_kamailio#1]
Hello, i want to make passthrough proxy via your expirience and now get this error.

Scheme is server with kamailio and rtpproxy have two network adapters: one local network and secont have white IP. All local routes goes to first network.

This logs when i try to register on my SMG from world.
XXX - white ip where kamailio listens 5060 port
YYY - white ip from where I try to register on XXX with softphone
10.125.184.2 - local ip of SMG(eltex sip device) nmap says it opened(btw it serv about 500 abons but now via ZTE b100 i need to replace HW with VM)

`Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/msg_parser.c:595]: parse_msg(): SIP Request:
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/msg_parser.c:597]: parse_msg():  method:  <SUBSCRIBE>
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/msg_parser.c:599]: parse_msg():  uri:     <sip:123456@XXX.XXX.XXX.XXX>
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/msg_parser.c:601]: parse_msg():  version: <SIP/2.0>
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/parse_via.c:1254]: parse_via_param(): Found param type 232, <branch> = <z9hG4bK807143de7202e911b8c4b7c09a168900>; state=6
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/parse_via.c:1254]: parse_via_param(): Found param type 235, <rport> = <n/a>; state=17
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/parse_via.c:2642]: parse_via(): end of header reached, state=5
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/msg_parser.c:486]: parse_headers(): Via found, flags=2
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/msg_parser.c:488]: parse_headers(): this is the first via
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [receive.c:178]: receive_msg(): After parse_msg...
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [receive.c:221]: receive_msg(): preparing to run routing scripts...
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=120 a=16 n=if
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=116 a=24 n=sanity_check
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/parse_addr_spec.c:871]: parse_addr_spec(): end of header reached, state=10
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/msg_parser.c:171]: get_hdr_field(): <To> [29]; uri=[sip:123456@XXX.XXX.XXX.XXX]
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: [97B blob data]
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/msg_parser.c:152]: get_hdr_field(): cseq <CSeq>: <2> <SUBSCRIBE>
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/msg_parser.c:183]: get_hdr_field(): content_length=0
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/msg_parser.c:89]: get_hdr_field(): found end of header
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: sanity [mod_sanity.c:248]: w_sanity_check(): sanity checks result: 1
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=127 a=16 n=if
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=120 a=25 n=mf_process_maxfwd_header
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: maxfwd [mf_funcs.c:85]: is_maxfwd_present(): value = 70
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=220 a=16 n=if
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=129 a=55 n=force_rport
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=179 a=16 n=if
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=157 a=16 n=if
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=135 a=24 n=loose_route
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: rr [loose.c:112]: find_first_route(): No Route headers found
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: rr [loose.c:944]: loose_route(): There is no Route HF
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=143 a=16 n=if
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=137 a=26 n=lookup_domain
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: domain [domain.c:230]: w_lookup_domain(): added AVP 'dattr_routeset'
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: domain [domain.c:255]: w_lookup_domain(): added AVP 'dattr_did'
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=151 a=16 n=if
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=155 a=16 n=if
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=151 a=33 n=ds_select_dst
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: dispatcher [dispatch.c:1837]: ds_select_dst_limit(): set [4001]
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/parse_addr_spec.c:171]: parse_to_param(): DEBUG: add_param: tag=1693815413
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [parser/parse_addr_spec.c:871]: parse_addr_spec(): end of header reached, state=29
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: dispatcher [dispatch.c:1948]: ds_select_dst_limit(): alg hash [3762732286]
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: dispatcher [dispatch.c:1990]: ds_select_dst_limit(): selected [1-4001/0] <sip:10.125.184.2:5060>
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=166 a=16 n=if
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=157 a=25 n=nat_uac_test
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [msg_translator.c:158]: check_via_address(): (YYY.YYY.YYY.YYY, YYY.YYY.YYY.YYY, 0)
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=166 a=24 n=add_path_received
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=167 a=25 n=rtpproxy_manage
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=168 a=24 n=record_route
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: *** cfgtrace:request_route=[DEFAULT_ROUTE] c=[/etc/kamailio/kamailio.cfg] l=222 a=1 n=forward
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [md5utils.c:67]: MD5StringArray(): MD5 calculated: 26aa9d95789ddaa54d6bd527a21d6118
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: <core> [forward.c:181]: get_out_socket(): no socket found
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: <core> [forward.c:183]: get_out_socket(): no corresponding socket found for(udp:10.125.184.2:5060)
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: ERROR: <core> [forward.c:527]: forward_request(): cannot forward to af 2, proto 1 no corresponding listening socket
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: WARNING: <core> [receive.c:241]: receive_msg(): error while trying script
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [receive.c:343]: receive_msg(): error:...
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [usr_avp.c:631]: destroy_avp_list(): destroying list 0x7ff18d7e0860
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [usr_avp.c:631]: destroy_avp_list(): destroying list (nil)
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [usr_avp.c:631]: destroy_avp_list(): destroying list (nil)
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [usr_avp.c:631]: destroy_avp_list(): destroying list (nil)
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [usr_avp.c:631]: destroy_avp_list(): destroying list (nil)
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [usr_avp.c:631]: destroy_avp_list(): destroying list (nil)
Dec 20 10:13:54 voip0 /usr/sbin/kamailio[32952]: DEBUG: <core> [xavp.c:446]: xavp_destroy_list(): destroying xavp list (nil)
`

This is domain table
`id(int,auto) domain(string) did(string,null) last_modified(int)
1:XXX.XXX.XXX.XXX:1:0
2:sip.customer02.com:2:0`

This is dispatcher table
`id(int,auto) setid(int) destination(string) flags(int) priority(int) attrs(string) description(string)
1:4001:sip\:10.125.184.2\:5060:0:1:backend=be011:"be011 default"
2:4002:sip\:10.100.0.18\:5060:0:1:backend=be021:"be021 default"
3:6001:sip\:[2001\:DB8\:8\:100\:\:17]\:5060:0:1:backend=be011:"be011 default"
4:6001:sip\:[2001\:DB8\:8\:100\:\:18]\:5060:0:1:backend=be021:"be021 default"
`

**Comment by denser** [comment]
I found glitch: kamailio wont listen on first interface but there is another app on 5060 port, so i add listen to local ip and 5080 port and its start working.

Now i get another ...thing.

I bad VOIP right now and just need to replace HW with SW ))) But cant clear vision about how kamailio works.

Before my SMG acc 123456 have sip doman 10.125.184.2 and IP on ZTE b100 local adress. Now I made registration works :) and realized that replace sip domain to XXX addres and ip addres(in smg filter on 123456) to YYY is not as it was. I need to replace data in communication packets to 10.125.184.2 and local ip of kamailio. Pls advice me how to do this with kamailio cfg?
